src/core/tests/type_prop/reshape.cpp

```diff
@@ -672,33 +667,550 @@ TEST(type_prop, reshape_dynamic_value_and_label_propagation) {
 
     const auto& et = element::i64;
     std::vector<int64_t> zero{0};
-    const auto indices = std::make_shared<op::v0::Constant>(et, Shape{}, zero);
-    const auto axis = std::make_shared<op::v0::Constant>(et, Shape{}, zero);
-    const auto gather = std::make_shared<op::v7::Gather>(shape_0, indices, axis);
+    const auto indices = make_shared<op::v0::Constant>(et, Shape{}, zero);
+    const auto axis = make_shared<op::v0::Constant>(et, Shape{}, zero);
+    const auto gather = make_shared<op::v7::Gather>(shape_0, indices, axis);
 
-    const auto output_pattern = std::make_shared<op::v0::Constant>(et, Shape{1}, std::vector<int64_t>{-1});
-    const auto unsqueeze = std::make_shared<op::v1::Reshape>(gather, output_pattern, false);
+    const auto output_pattern = make_shared<op::v0::Constant>(et, Shape{1}, std::vector<int64_t>{-1});
+    const auto unsqueeze = make_shared<op::v1::Reshape>(gather, output_pattern, false);
 
-    auto bc = std::make_shared<op::v1::Broadcast>(param, unsqueeze);
+    auto bc = make_shared<op::v1::Broadcast>(param, unsqueeze);
     ASSERT_EQ(bc->get_shape(), (Shape{3}));
 
     const auto& output_shape = bc->get_output_partial_shape(0);
-    ASSERT_EQ(ov::DimensionTracker::get_label(output_shape[0]), 10);
+    EXPECT_EQ(output_shape, PartialShape({3}));
+    EXPECT_THAT(get_shape_labels(output_shape), ElementsAre(10));
 }
 
-TEST(type_prop, reshape_label_shape_propagation_minus_one) {
-    Dimension marked_0 = Dimension(-1);
-    ov::DimensionTracker::set_label(marked_0, 10);
+TEST(type_prop, reshape_when_pattern_has_static_shape_only) {
+    auto param = make_shared<op::v0::Parameter>(element::f32, Shape{3, 4});
+    auto shape_pattern = make_shared<op::v0::Parameter>(element::u64, PartialShape{3});
+    auto r = make_shared<op::v1::Reshape>(param, shape_pattern, false);
+
+    EXPECT_EQ(r->get_element_type(), element::f32);
+    EXPECT_EQ(r->get_output_partial_shape(0), PartialShape::dynamic(3));
+}
+
+TEST(type_prop, reshape_when_pattern_has_interval_shape_only) {
+    auto param = make_shared<op::v0::Parameter>(element::f32, Shape{3, 4});
+    auto shape_pattern = make_shared<op::v0::Parameter>(element::u64, PartialShape{{1, 3}});
+    auto r = make_shared<op::v1::Reshape>(param, shape_pattern, false);
+
+    EXPECT_EQ(r->get_element_type(), element::f32);
+    EXPECT_EQ(r->get_output_partial_shape(0), PartialShape::dynamic());
+}
+
+TEST(type_prop, reshape_when_pattern_has_scalar_shape_only) {
+    auto param = make_shared<op::v0::Parameter>(element::f32, Shape{3, 4});
+    auto shape_pattern = make_shared<op::v0::Parameter>(element::u64, PartialShape{});
+    auto r = make_shared<op::v1::Reshape>(param, shape_pattern, false);
+
+    EXPECT_EQ(r->get_element_type(), element::f32);
+    EXPECT_EQ(r->get_output_partial_shape(0), PartialShape());
```

Comment:
I think in this test the output rank should be {0,1}, because value of scalar is unknown.
But if scalar value will be constant and 1 then input shape never become sacalar and shape inference can throw exception.

Reply:
Done.